corehq/apps/userreports/indicators/factory.py

```diff
@@ -85,6 +87,42 @@ def _construct_column(choice):
     return CompoundIndicator(base_display_name, choice_indicators)
 
 
+def _build_ledger_balances_indicator(spec, context):
+    wrapped_spec = LedgerBalancesIndicatorSpec.wrap(spec)
+    base_display_name = wrapped_spec.display_name
+
+    def _construct_display(product_code):
+        return '{base} ({product_code})'.format(base=base_display_name,
+                                                product_code=product_code)
+
+    def _construct_column(product):
+        return '{col}_{product}'.format(col=spec['column_id'], product=product)
+
+    def _make_get_stock_state(product_code):
+        def get_stock_state(doc, context):
+            try:
+                return StockState.objects.get(
+                    section_id=wrapped_spec.ledger_section,
+                    case_id=doc['_id'],
```

Comment:
I think it would be nice if this was an expression, so that you could link a ledger row to something other than the primary case table containing that ledger.